### ISSUE TITLE
[Feature Request] Add support for single instance loading

### ISSUE THREAD

**Issue by Ohmarinus** [SoftFever/OrcaSlicer#817]
**Is your feature request related to a problem? Please describe.**
The problem is that when an STL is exported from Fusion360 to OrcaSlicer under windows, it always opens up a new instance of OrcaSlicer.

With PrusaSlicer the fix is to run it via a bat with the --single-instance affix in the command line. Sadly, this option does not seem to exist for OrcaSlicer. It also works for Cura. So I would really like to see this option in OrcaSlicer.

**Describe the solution you'd like**
Adding the option to add --single-instance in the command line will bring a lot of improvement to the firmware.

**Describe alternatives you've considered**
I have tried adding this line in the same way I do with PrusaSlicer and Cura, but for OrcaSlicer it doesn't appear to work.

**Additional context**
What I personally do is create a *.bat file and call that file from Fusion360, this works the best.
The file contents would look like this:
`@start "" "%~dp0\orca-slicer.exe" --single-instance %* &`

**Comment by noobydp** [comment]
+1 

Superslicer and Cura have Preferences in the app to control this behaviour, this would work better for me than the command line switch.

![image](https://github.com/SoftFever/OrcaSlicer/assets/5045555/e3b03ff3-eb69-4850-90a9-78011a67eefc)

**Comment by Ohmarinus** [comment]
> +1
> 
> Superslicer and Cura have Preferences in the app to control this behaviour, this would work better for me than the command line switch.

Yes, having them as a setting in-app would also fix the issue. Although I do expect the solution that PrusaSlicer uses is super easy to implement since it already exists and Orca is a forked version. I was actually surprised to find this command-line option didn't exist in OrcaSlicer.

**Comment by arraylabs** [comment]
I would some this feature as well, working on a design I end up with a pile or orcaslicer windows opened which isn't ideal. Thanks!

**Comment by arraylabs** [comment]
Here is a workaround in the meantime:
```
@echo off
setlocal

:: Define the name of the process.
set "appPath=C:\Users\USERNAME\Documents\OrcaSlicer_Current\orca-slicer.exe"
set "appName=orca-slicer.exe"

:: Check if the process is running.
tasklist /FI "IMAGENAME eq %appName%" 2>NUL | find /I /N "%appName%">NUL
if not "%ERRORLEVEL%"=="0" (
    start "" "%appPath%"
)

endlocal
```

**Comment by julie777** [comment]
> > +1
> > Superslicer and Cura have Preferences in the app to control this behaviour, this would work better for me than the command line switch.
> 
> Yes, having them as a setting in-app would also fix the issue. Although I do expect the solution that PrusaSlicer uses is super easy to implement since it already exists and Orca is a forked version. I was actually surprised to find this command-line option didn't exist in OrcaSlicer.

The preference was already in Prusa Slicer when BBL forked it. I wonder if they deleted all the code, or just removed all the settings from preferences.

**Comment by arraylabs** [comment]
From a very quick look at prusaslicer code vs orcaslicer code it looks like the preferences were removed somewhere along the way. The slicer3d stuff still appears to be intact.

**Comment by jbkuma** [comment]
+1 on this, it's sort of a critical feature and I already miss using Prusa Slicer now that I've picked up the P1P!

**Comment by Senkii-code** [comment]
+1, loading a new instance every time one exports a model is unnecessarily time consuming. Ideal would be an option to load to existing or new plate.

**Comment by npopson** [comment]
+1, I miss this feature from PrusaSlicer.

**Comment by AndBu** [comment]
Bump.

**Comment by ipd-otto** [comment]
> +1, it´s really annoying to export models from fusion...

I've got the same problem. It'd be a much better workflow with single instance.

**Comment by lucas577** [comment]
I was kinda disappointed when it opened a new instance of orca, I hope this get implemented aswell. +1

**Comment by Lincoln-Jones** [comment]
Still very much hoping this can get addressed. Being able to export a file from Fusion and have it open in an existing OrcaSlicer instance would be incredibly convenient.

**Comment by a0s** [comment]
So, how to override this in dirty way under mac?

**Comment by gbumgard** [comment]
I just had to switch to using a Windows PC to run Fusion after my MacBook died. Starting a new instance of Orca every export is annoying. Definitely works better under MacOS where I can send as many bodies as I want to the same plate.

**Comment by alexeyzel** [comment]
I moved to Orca from PrusaSlicer and sending files for print from Fusion to Orca is really annoying because each time it opens a new window. +1 for this feature request.

**Comment by Ohmarinus** [comment]
> I moved to Orca from PrusaSlicer and sending files for print from Fusion to Orca is really annoying because each time it opens a new window. +1 for this feature request.

I'm surprised that after a whole year not a single DEV has reacted to this question as I think it's a very valuable addition. I am still missing the function quite a lot.

For now my workaround is to directly export my entire project as a STEP file from Fusion360 and then import the entire design and pick the parts that I want to print. But being able to force Orca into a single instance is so much better.

**Comment by Ohmarinus** [comment]
> It seems that the best solution here would be to open the file on a new "Plate" in the same project in the same application instance. To a first approximation, OrcaSlicer should be able to receive signals one way or another about opening files (dbus is not in MacOS - how do other programs do this?)

I'm not too experienced with these things but I work on Windows 10 and it seems that this is how it works in Windows:
https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-instancing

**Comment by ruddy17** [comment]
This is a must-have feature! Also have been waiting for it since day 1 of transitioning to orca. It is implemented in all other slicers, including prusa, which orca was forked from. Duh.

**Comment by noobydp** [comment]
This will be resolved by #4810
